### PR TITLE
Docs add capabilities section

### PR DESCRIPTION
**Pull request by peteroneilljr**
Adding in instructions to the OPA docs on how to use capabilities feature. 

Resolves: https://github.com/open-policy-agent/opa/issues/3769

Signed-off-by: Peter ONeill <peteroneilljr@gmail.com>

### CODE REVIEW

File: docs/content/deployments.md [hunk 1]
```diff
@@ -364,3 +365,119 @@ The mechanisms discussed above ensure that OPA is not asked to answer policy que
...
 
 In Kubernetes admission control, for example, the Kubernetes admin can choose whether to fail-open or fail-closed, leaving the decision up to the user.  And often this is the correct way to build an integration because it is unlikely that there is a universal solution.  For example, running an OPA-integration in a development environment might require fail-open, but running exactly the same integration in a production environment might require fail-closed.
+
+## Capabilities
+
+OPA now supports a _capabilities_ check on policies. The check allows callers to restrict the [built-in](../kubernetes-primer/#builtins) functions that policies may depend on. If the policies passed to OPA require built-ins not listed in the capabilities structure, an error is returned. The capabilities check is currently supported by the `check` and `build` sub-commands and can be accessed programmatically on the `ast.Compiler` structure. The OPA repository includes a set of capabilities files for previous versions of OPA in the [capabilities](https://github.com/open-policy-agent/opa/tree/main/capabilities) folder.
```
Comment (srenatus) [inline docs/content/deployments.md:371]: Is that first link correct? I'd expect `../policy-reference/#built-in-functions` and double-check it from the netlify preview 🙃

**Review by srenatus — APPROVED** [review]
LGTM.

There's also some stuff about wasm ABI major/minor versions in the capabilities file, but what you've added here is infinitely better than what we had before. 🚀

File: docs/content/deployments.md (hunk 1 above)
Comment (peteroneilljr) [reply docs/content/deployments.md:371]: Hmm it is a valid link, but your link is probably the correct one for this use case.

Comment (peteroneilljr) [reply docs/content/deployments.md]: updated

**Comment by peteroneilljr** [comment]
Signed-off-by: Peter ONeill peteroneilljr@gmail.com

**Pull request merged**